internal/api/client.go

```diff
@@ -17,25 +20,50 @@ type ClientConfig struct {
 	AccessToken      string
 	OrganizationSlug string
 	ServerBaseUrl    string
+	DebugEnabled     bool
 }
 
 // authTransport is a middleware for the HTTP client.
 type authTransport struct {
-	accessToken string
+	accessToken  string
+	debugEnabled bool
 }
 
 // RoundTrip adds the Authorization header to all requests made by the HTTP client.
 func (t *authTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if t.debugEnabled {
+		fmt.Println("--- 🕵️ DEBUG")
+		fmt.Println("Request:")
+		fmt.Printf("%s %s\n\n", req.Method, req.URL.String())
```

Comment:
You don't need to manually call the `String` method when using `fmt.Printf` - just use `%v`.